Convert SPI intel driver to DTS

tbursztyka opened the pull request.

Commit:
drivers/spi: Always selecet HAS_DTS_SPI once SPI is enabled

All drivers require DTS for their primary SPI settings.

Removing SPI_[0-9]_NAME config option added some more samples changes.
Usage of these options there was anyway not relevant.

Fixes #11064

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
There is a chicken & egg issue: if I want to remove SPI_0_NAME from drivers/spi/Kconfig, I should first make cc2520 (ieee802154) dts ready. Will do that, and make this current PR on top of that 15.4 change branch.

Commit:
dts/boards: Add a dedicated cc2520 config for quark_se_c1000_devboard

Using DTS to setup cc2520 SPI settings on quark_se_c1000_devboard.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
myeah there is even more SPI name config issue all over the samples, which means more cleanup there.

galak commented on dts/bindings/spi/intel,intel-spi.yaml:
```diff
@@ -0,0 +1,31 @@
+
+properties:
+    compatible:
+      constraint: "intel,intel-spi"
```
Is there anything more detailed or descriptive than 'intel,intel-spi'?  Seems a bit too generic

galak replied:
Ok, maybe we should but a bit more description  in 'description:' field about what SoC this applies to right now.

galak reviewed (changes requested):
Looks good beyond the compatible feeling a bit too generic.

galak commented:
> myeah there is even more SPI name config issue all over the samples, which means more cleanup there.

Yeah, there is Issue #10894 that covers all the SPI slave drivers that need converting over.  Let's get the controller stuff merged and than we can tackle the slave drivers.

tbursztyka commented:
Let's try this quick fix for cc2520, hopefully that will pass now.

Commit:
boards/x86: Use dts for configuring the cc2520 device

Only the SPI bus for now.
Make cc2520's Kconfig aware of DTS on these settings.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

galak commented on drivers/ieee802154/Kconfig.cc2520:
```diff
@@ -14,18 +14,21 @@ if IEEE802154_CC2520
 
 config IEEE802154_CC2520_DRV_NAME
 	string "TI CC2520 Driver's name"
+	depends on !HAS_DTS_SPI
```
Under what situation after this PR will be able to set !HAS_DTS_SPI?  Seems like you should just remove this.

galak commented on boards/x86/quark_se_c1000_devboard/quark_se_c1000_devboard.dts:
```diff
@@ -78,4 +78,5 @@
 
 &spi1 {
 	status = "ok";
+
```
Is this missing the cc2520 node?

galak commented on dts/bindings/spi/intel,intel-spi.yaml:
```diff
@@ -0,0 +1,31 @@
+version: 0.1
+
+description: >
+     This is a representation of the Intel SPI controller.
```
We need to add a bit more than just 'Intel SPI Controller' so someone has a bit of clue when to use this binding/compatible.

tbursztyka replied:
Why should this binding be more precise while other are exactly the same? (I can add galileo in the mix, but that's not really the place)

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -166,7 +166,7 @@
 				label = "cc2520";
 				spi-max-frequency = <8000000>;
-				status = "disabled";
+				status = "ok";
```
Why this change here?  Typically done in the board.dts

tbursztyka replied:
that's due to a wrong rebase, move it to proper commit

galak replied:
? shouldn't the addition of the cc2520@0 node & cs-gpios be in the board.dts.  Might still be a rebase issue.

tbursztyka replied:
no you'r right, cc2520 is present on the board, not on the soc

galak reviewed (changes requested):
* Some DTS cleanup/changes
* Also not sure what's going on w/the changes to the samples/drivers/led_ws2812 and samples/drivers/led_lpd8806?

tbursztyka commented:
@galak about the led_strips samples, these were uselessly renaming spi controller name in order no to some irrelevant "double check" on spi controller being up, when it's all about led strip (if it's badly configured, the sample won't work and that's it).

dbkinder reviewed (approved):
doc changes LGTM

tbursztyka commented:
@galak pleate, re-review.

tbursztyka commented:
@galak  So you forked only part of the PR? That's even more not a proper way. What is this process of working ...

galak commented:
> @galak So you forked only part of the PR? That's even more not a proper way. What is this process of working ...

We'll it annoys me that github doesn't let me merge parts of a PR, so from time to time I do this so I can stop reviewing the same commits over and over again.  Sorry for not explain that in more detail.
